Mage.Sets/src/mage/cards/g/GluttonousHellkite.java

```diff
@@ -0,0 +1,170 @@
+
+        List<Permanent> sacPermanents = new ArrayList<>();
+        Player controller = game.getPlayer(source.getControllerId());
+        Object obj = getValue(CastSourceTriggeredAbility.SOURCE_CAST_SPELL_ABILITY);
+        int sacrificeAmount = 0;
+        if (controller == null ) {
+            return false;
+        }
+        if (obj instanceof SpellAbility) {
+            sacrificeAmount = ((SpellAbility) obj).getManaCostsToPay().getX();
+        }
+        if (sacrificeAmount < 1) {
+            return true;
+        }
+        for (UUID playerId : game.getState().getPlayersInRange(controller.getId(), game)) {
+            Player player = game.getPlayer(playerId);
+            if (player != null) {
+                TargetSacrifice target = new TargetSacrifice(sacrificeAmount, StaticFilters.FILTER_PERMANENT_CREATURE);
+                if (target.canChoose(player.getId(), source, game)) {
+                    while (!target.isChosen(game) && target.canChoose(player.getId(), source, game) && player.canRespond()) {
+                        player.choose(Outcome.Sacrifice, target, source, game);
+                    }
+                    target.getTargets().stream().map(game::getPermanent).filter(Objects::nonNull).forEach(sacPermanents::add);
+                }
+            }
+        }
+        int counters = sacPermanents
+                .stream()
+                .mapToInt(p -> p.sacrifice(source, game) ? 2 : 0)
+                .sum();
+
+
+        for (Effect effect : source.getEffects()) {
+            if (effect instanceof GluttonousHellkiteReplacementEffect) {
+                effect.setValue("GluttonousHellkiteCounters", counters);
+            }
+        }
+        return true;
+    }
+}
+
+class GluttonousHellkiteReplacementEffect extends ReplacementEffectImpl {
+
+    GluttonousHellkiteReplacementEffect() {
+        super(Duration.OneUse, Outcome.BoostCreature);
+        staticText = "{this} enters the battlefield with two +1/+1 counters on it for each creature sacrificed this way";
+    }
+
+    private GluttonousHellkiteReplacementEffect(final GluttonousHellkiteReplacementEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean checksEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.ENTERS_THE_BATTLEFIELD;
+    }
+
+    @Override
+    public boolean applies(GameEvent event, Ability source, Game game) {
+        return event.getTargetId().equals(source.getSourceId());
+    }
+
+    @Override
+    public boolean replaceEvent(GameEvent event, Ability source, Game game) {
+
+        Object object = this.getValue("GluttonousHellkiteCounters");
+        int amount = ((Integer) object);
+        ((EntersTheBattlefieldEvent) event).getTarget().addCounters(CounterType.P1P1.createInstance(amount), source.getControllerId(), source, game);
+            game.getBattlefield().addPermanent(((EntersTheBattlefieldEvent) event).getTarget());
+
+        return true;
```

Comment:
this is wrong here - for replaceEvent, you only return true if you consume the event and nothing else should happen. I assume that's why you are manually adding the permanent, but correct way is to not do that, return false, and let the event continue being handled

see Naya Soulbeast for example (also, further evidence that a custom effect is indeed appropriate).